pkg/capabilities/mercury/mercury.go

```diff
@@ -0,0 +1,31 @@
+package mercury
+
+import "github.com/smartcontractkit/chainlink-common/pkg/values"
+
+type FeedID = uint64
+
+type ReportSet struct {
+	// feedID -> report
+	Reports map[FeedID]Report
+}
+
+type Report struct {
+	Info       ReportInfo // minimal data extracted from the report for convenience
+	FullReport []byte     // full report, acceptable by the verifier contract
+}
+
+type ReportInfo struct {
+	Timestamp uint32
+	Price     float64
+}
+
+//go:generate mockery --quiet --name Codec --output ./mocks/ --case=underscore
+type Codec interface {
```

Comment:
IMO this is in the wrong place: it's considered idiomatic to declare interfaces where they are used, not in the "source" package. This is something we get wrong a lot at CLL and causes problems as we then introduce dependencies on the package hosting the concrete struct (admittedly that's not the case here).

Reply:
Where would you put it then? This is a Mercury Codec that can be used by multiple places in the code.

Reply:
Moving to datafeeds.